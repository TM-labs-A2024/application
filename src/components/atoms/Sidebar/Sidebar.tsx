import { ChevronLeftIcon } from '@chakra-ui/icons'
import {
  Drawer,
  DrawerBody,
  DrawerFooter,
  DrawerHeader,
  DrawerOverlay,
  DrawerContent,
  DrawerCloseButton,
  IconButton
} from '@chakra-ui/react'
import React, { ReactNode } from 'react'

export default function Sidebar({
  isOpen,
  onClose,
  children,
  innerRef
}: {
  isOpen: boolean
  onClose: () => void
  children: ReactNode
  innerRef: React.RefObject<HTMLButtonElement>
}) {
  return (
    <Drawer isOpen={isOpen} placement="left" onClose={onClose} finalFocusRef={innerRef}>
      <DrawerOverlay />
      <DrawerContent>
        <DrawerCloseButton />
        <DrawerHeader>HealthCore</DrawerHeader>

        <DrawerBody>{children}</DrawerBody>

        <DrawerFooter>
          <IconButton
            size="xl"
            fontSize="30px"
            aria-label="back"
            variant="link"
            icon={<ChevronLeftIcon />}
            onClick={onClose}
          />
        </DrawerFooter>
      </DrawerContent>
    </Drawer>
  )
}
