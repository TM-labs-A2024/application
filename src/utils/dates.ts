export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
