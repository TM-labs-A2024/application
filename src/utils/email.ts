import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
