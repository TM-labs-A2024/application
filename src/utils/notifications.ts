import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
