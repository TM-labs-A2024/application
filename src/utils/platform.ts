import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
