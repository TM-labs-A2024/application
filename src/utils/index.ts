import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
