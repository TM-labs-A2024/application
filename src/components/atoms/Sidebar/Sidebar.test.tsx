import { render, screen } from '@test/utils/index'
import react from 'react'

import '@testing-library/jest-dom'
import Sidebar from './Sidebar'

const props = {
  isOpen: true,
  onClose: () => null,
  innerRef: react.createRef<HTMLButtonElement>()
}

const invisibleProps = {
  ...props,
  isOpen: false
}

const component = (
  <Sidebar {...props}>
    <div>Child</div>
  </Sidebar>
)

const componentNotVisible = (
  <Sidebar {...invisibleProps}>
    <div>Child</div>
  </Sidebar>
)

describe('Atoms > Sidebar test', () => {
  test('The component is visible when the isOpen prop is true', () => {
    render(component)

    expect(screen.getByRole('dialog')).toBeVisible()
  })

  test('The component is not visible when the isOpen prop is false', () => {
    render(componentNotVisible)

    expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
  })

  test('The children of the component are the expected ones', async () => {
    render(component)

    expect(await screen.findByText('Child')).toBeInTheDocument()
  })

  test('Matches the snapshot', async () => {
    render(component)

    expect(await screen.findByRole('dialog')).toMatchSnapshot()
  })
})
