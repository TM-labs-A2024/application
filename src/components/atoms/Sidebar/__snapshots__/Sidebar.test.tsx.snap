// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Sidebar test Matches the snapshot 1`] = `
<div
  aria-describedby="chakra-modal--body-:r9:"
  aria-labelledby="chakra-modal--header-:r9:"
  aria-modal="true"
  class="chakra-slide chakra-modal__content css-1o3pyl4"
  id="chakra-modal-:r9:"
  role="dialog"
  style="position: fixed; left: 0px; top: 0px; bottom: 0px; width: 100%; transform: none;"
  tabindex="-1"
>
  <button
    aria-label="Close"
    class="chakra-modal__close-btn css-1ik4h6n"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <header
    class="chakra-modal__header css-9fgtzh"
    id="chakra-modal--header-:r9:"
  >
    Taco Medical
  </header>
  <div
    class="chakra-modal__body css-79z5gx"
    id="chakra-modal--body-:r9:"
  >
    <div>
      Child
    </div>
  </div>
  <footer
    class="chakra-modal__footer css-k0waxj"
  >
    <button
      aria-label="back"
      class="chakra-button css-wpzvz6"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          fill="currentColor"
        />
      </svg>
    </button>
  </footer>
</div>
`
